### PR TITLE
Loader for Salsa instrument at ILL

### PR DESCRIPTION
**Pull request by gui-co**
**Description of work.**

This PR adds a loader for SALSA instrument from the ILL.

**To test:**

Code review and try to load this file: [046430.zip](https://github.com/mantidproject/mantid/files/8462949/046430.zip)

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/DataHandling/src/LoadILLSALSA.cpp [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+void LoadILLSALSA::setInstrument(double distance, double angle) {
+  // load instrument
+  auto loadInst = createChildAlgorithm("LoadInstrument");
+  loadInst->setPropertyValue("InstrumentName", "SALSA");
+  loadInst->setProperty<API::MatrixWorkspace_sptr>("Workspace", m_outputWorkspace);
+  loadInst->setProperty("RewriteSpectraMap", Kernel::OptionalBool(true));
+  loadInst->execute();
+  setProperty("OutputWorkspace", m_outputWorkspace);
+
+  // translation
```
Comment (darominski) [inline Framework/DataHandling/src/LoadILLSALSA.cpp:125]: Do you think it would make sense to extract placing of the instrument to own function?

**Review by darominski — CHANGES_REQUESTED** [review]
This looks fine, a couple of small questions indispersed with the code. The attached data loads perfectly fine.

There's still a matter of no-scan numors with `Multi_data` type and whether this should be supported on that stage.

Could you add a unit test for the loader and a SALSA test to `LoadILLTest`?

File: Framework/DataHandling/src/LoadILLSALSA.cpp (hunk 1 above)
Comment (gui-co) [reply Framework/DataHandling/src/LoadILLSALSA.cpp:125]: I set the instrument (loading + position) in a dedicated function: `LoadILLSALSA::setInstrument()`

**Comment by gui-co** [comment]
I added the tests in `LoadILLTests.h`.

**Review by darominski — CHANGES_REQUESTED** [review]
Thanks for all of the changes, there is no issue now loading the provided data and also data from two previous cycles I could find. It's good to see the confidence test as well.

Could you provide a couple of unit tests for the loader, e.g. V1, V2, and a performance one?

File: Framework/DataHandling/src/LoadILLSALSA.cpp [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+  H5::DataSpace detectorDataspace = detectorDataset.getSpace();
+
+  int nDims = detectorDataspace.getSimpleExtentNdims();
+  std::vector<hsize_t> dimsSize(nDims);
+  detectorDataspace.getSimpleExtentDims(dimsSize.data(), NULL);
+
+  size_t numberOfScans = dimsSize[0];
+
+  if ((dimsSize[1] != VERTICAL_NUMBER_PIXELS) || (dimsSize[2] != HORIZONTAL_NUMBER_PIXELS))
+    throw std::runtime_error("");
```
[the remark this answers is not in the document]
Comment (gui-co) [reply Framework/DataHandling/src/LoadILLSALSA.cpp]: I improved the error message

**Comment by gui-co** [comment]
Thanks for the review. I added the missing unit test/performance test.

**Review by darominski — APPROVED** [review]
Thanks for the changes. It's nice to see performance and confidence tests. Looks ready to go.

**Pull request merged**